File: java/src/com/liferay/nativity/modules/fileicon/win/WindowsFileIconControlImpl.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.nativity.modules.fileicon.win;

import com.liferay.nativity.Constants;
import com.liferay.nativity.control.NativityControl;
import com.liferay.nativity.control.win.WindowsNativityUtil;
import com.liferay.nativity.modules.fileicon.FileIconControlBase;
import com.liferay.nativity.modules.fileicon.FileIconControlCallback;
import com.liferay.nativity.util.win.RegistryUtil;

import java.util.Map;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
* @author Dennis Ju
*/
public class WindowsFileIconControlImpl extends FileIconControlBase {

	public WindowsFileIconControlImpl(
		FileIconControlCallback fileIconControlCallback,
		NativityControl nativityControl) {

		super(fileIconControlCallback, nativityControl);
	}

	@Override
	public void disableFileIcons() {
		RegistryUtil.writeRegistry(
			Constants.NATIVITY_REGISTRY_KEY,
			Constants.ENABLE_OVERLAY_REGISTRY_NAME, 0);
	}

	@Override
	public void enableFileIcons() {
		RegistryUtil.writeRegistry(
			Constants.NATIVITY_REGISTRY_KEY,
			Constants.ENABLE_OVERLAY_REGISTRY_NAME, 1);
	}

	@Override
	public void refreshIcons() {
	}

	@Override
	public void refreshIcons(String[] paths) {
		if (!WindowsNativityUtil.loaded()) {
			return;
		}

		if ((paths == null) || (paths.length == 0)) {
			return;
		}

		try {
			for (String path : paths) {
				WindowsNativityUtil.updateExplorer(path);
			}
		}
		catch (UnsatisfiedLinkError ule) {
			_logger.error(ule.getMessage(), ule);
		}
	}

	@Override
	public void refreshWindow(String path) {
		if (!WindowsNativityUtil.loaded()) {
			return;
		}

		WindowsNativityUtil.refreshExplorer(path);
	}

	@Override
	public int registerIcon(String path) {
		return 0;
	}

	@Override
	public void registerIconWithId(String path, String label, String iconId) {
	}

	@Override
	public void removeAllFileIcons() {
	}

	@Override
	public void removeFileIcon(String path) {
	}

	@Override
	public void removeFileIcons(String[] paths) {
	}

	@Override
	public void setFileIcon(String path, int iconId) {
	}

	@Override
	public void setFileIcons(Map<String, Integer> fileIconsMap) {
	}

	@Override
	public void unregisterIcon(int id) {
	}

	private static Logger _logger = LoggerFactory.getLogger(
		WindowsFileIconControlImpl.class.getName());

}

File: csharp/src/Liferay.Nativity/Modules/FileIcon/Win/WindowsFileIconControlImpl.cs
/**
 * Copyright (c) 2000-2013 Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Liferay.Nativity.Control;
using Microsoft.Win32;

namespace Liferay.Nativity.Modules.FileIcon.Win
{
	/**
	* @author Dennis Ju
	* @author Patryk Strach - C# port
	*/
	public class WindowsFileIconControlImpl : FileIconControlBase
	{
		public WindowsFileIconControlImpl(NativityControl nativityControl, FileIconControlCallback fileIconControlCallback)
			: base(nativityControl, fileIconControlCallback)
		{
		}

		public override void DisableFileIcons()
		{
			Registry.SetValue(Constants.NATIVITY_REGISTRY_KEY, Constants.ENABLE_OVERLAY_REGISTRY_NAME, "0");
		}

		public override void EnableFileIcons()
		{
			Registry.SetValue(Constants.NATIVITY_REGISTRY_KEY, Constants.ENABLE_OVERLAY_REGISTRY_NAME, "1");
		}

		public override void RefreshIcons()
		{
		}

		public override int RegisterIcon(string path)
		{
			return 0;
		}

		public override int RegisterMenuIcon(string path)
		{
			return 0;
		}

		public override void RemoveAllFileIcons()
		{
		}

		public override void RemoveFileIcon(string path)
		{
		}

		public override void RemoveFileIcons(IEnumerable<string> paths)
		{
		}

		public override void SetFileIcon(string path, int iconId)
		{
		}

		public override void SetFileIcons(IDictionary<string, int> fileIconsMap)
		{
		}

		public override void UnregisterIcon(int id)
		{
		}
	}
}
